### ISSUE TITLE
OverflowError in find_introns

### ISSUE THREAD

**Issue by ksahlin** [pysam-developers/pysam#650]
Hi, 

Thanks for a great library! I get the following error when calling `samfile.find_introns(samfile.fetch(until_eof=True))`. This only happens on some of my datasets. I use pysam 0.14.1. Let me know if you want a sample file to fix this bug.

```
Traceback (most recent call last):
  File "filter_hits_on_hg19.py", line 321, in <module>
    main(args)
  File "filter_hits_on_hg19.py", line 301, in main
    all_matches, alignment_file = SAM_filter(args.samfile, ALIGNMENT_START, ALIGNMENT_END, ALIGN_COVERAGE, ALIGN_IDENTITY)
  File "filter_hits_on_hg19.py", line 246, in SAM_filter
    introns = SAM_file.find_introns(SAM_file.fetch(until_eof=True))
  File "pysam/libcalignmentfile.pyx", line 1632, in pysam.libcalignmentfile.AlignmentFile.find_introns (pysam/libcalignmentfile.c:19365)
OverflowError: can't convert negative value to uint32_t
```

**Comment by AndreasHeger** [comment]
Thanks, will check.

**Comment by ksahlin** [comment]
Hi @wckdouglas,

I produced the header to the SAM-files myself, and the alignment fields were generated from a third party script. After various updates to my pipeline (such as the generating the header) I cannot reproduce the error in any of my files. It could be that I formatted the headers incorrectly. 

I think we can close this issue but I will reopen if I encounter this error again at some point. 

Thanks!
